src/tree/mod.rs

```diff
@@ -706,7 +716,20 @@ impl RatchetTree {
         // Determine if Commit needs a path field
         let path_required = has_updates || has_removes || !has_adds;
 
-        Ok((path_required, self_removed, invitation_list))
+        // Collect the new leaves indexes so we can filter them out in the resolution later
+        let new_leaves_indexes: HashSet<NodeIndex> = HashSet::from_iter(
+            invitation_list
```

Comment:
The question is though whether we really need to return this here. `new_leaves_indexes` can be computed from the return value `invitation_list`. It has to happen somewhere, it's just that there are already so many return values here.